src/main/java/uk/gov/hmcts/reform/ccd/document/am/controller/endpoints/CaseDocumentAmController.java

```diff
@@ -239,4 +242,42 @@ public CaseDocumentAmController(ObjectMapper objectMapper, HttpServletRequest re
             throw new ResponseFormatException("Exception while uploading the documents :" + e);
         }
     }
+
+    @Override
+    public ResponseEntity<Object> generateHashCode(
+        @ApiParam(value = Constants.S2S_API_PARAM, required = true)
+        @RequestHeader(value = Constants.SERVICE_AUTHORIZATION, required = true) String serviceAuthorization,
+
+        @ApiParam("Authorization header of the currently authenticated user")
+        @RequestHeader(value = "Authorization", required = true) String authorization,
+
+        @ApiParam("documentId")
+        @PathVariable("documentId") UUID documentId,
+
+        @ApiParam(value = "CaseType identifier for the case document.", required = true)
+        @NotNull(message = "Provide the Case Type ID ")
+        @RequestHeader(value = "caseTypeId", required = true) String caseTypeId,
+
+        @ApiParam(value = "Jurisdiction identifier for the case document.", required = true)
+        @NotNull(message = "Provide the Jurisdiction ID ")
+        @RequestHeader(value = "jurisdictionId", required = true) String jurisdictionId) {
+
+        try {
+            ValidationService.validateInputParams(INPUT_STRING_PATTERN, documentId.toString(), caseTypeId, jurisdictionId);
+
+            HashMap<String, String> responseBody = new HashMap<>();
+
+            String hashedToken = ApplicationUtils.generateHashCode(documentId.toString().concat(jurisdictionId).concat(caseTypeId));
+            responseBody.put(HASHCODE, hashedToken);
```

Comment:
The hash-code key is picked from the constants to be consistent as per upload API